google/cloud/pubsub/internal/batch_callback.h

```diff
@@ -44,6 +44,9 @@ class BatchCallback {
   virtual void message_callback(MessageCallback::ReceivedMessage m) = 0;
   virtual void user_callback(MessageCallback::MessageAndHandler m) = 0;
 
+  virtual void AddEvent(std::string const& ack_id,
```

Comment:
It's not too late! If you take my suggestion you can add these events one at a time:

e.g., start with `ModAck(std::string const& ack_id)`

Reply:
Done for all